### PR TITLE
Fix path filters

### PR DESCRIPTION
**Pull request by mitchdenny**
Due to my ignorance I didn't realize that path filters in Azure Pipelines (when defined in YAML) don't patch partial path segments. So we had some instances where we had an exclude filter such as ```sdk/keyvault/microsoft-azure-``` assuming it would exclude changes in ```sdk/keyvault/microsoft-azure-keyvault``` when in fact it won't.

Unfortunately that means that we need to explicitly specify the full path segment for the directories that we want to exclude. That is what this PR is.

### CODE REVIEW

File: sdk/search/ci.mgmt.yml [hunk 1]
```diff
@@ -15,9 +15,7 @@ trigger:
     include:
       - sdk/search/
     exclude:
-      - sdk/search/azure
-      - sdk/search/microsoft-azure
-      - sdk/search/ms-azure
+      - sdk/search/azure-search
```
Comment (sima-zhu) [inline sdk/search/ci.mgmt.yml:18]: Search rename the model to azure-search-documents

Comment (mitchdenny) [reply sdk/search/ci.mgmt.yml:18]: Fixed.

**Comment by weshaggard** [comment]
This seems like something we might need to repeat sometimes and I suspect you probably wrote a script to do it and if so can you check that script in so we can update these easier in the future?

**Comment by mitchdenny** [comment]
I didn't script it (much) I just had some PowerShell that generated the include lines. I think it would be possible to automate however. But its probably a discrete task. I think we already have an issue to automate (partially) the creation of ci.yml files.

**Pull request merged**